cmake/ConfigureCompiler.cmake

```diff
@@ -217,6 +217,12 @@ else()
       -Wno-undefined-var-template
       -Wno-tautological-pointer-compare
       -Wno-format)
+    set(USE_AVX ON CACHE BOOL "Enable AVX instructions")
```

Comment:
We know that these will affect correctness (an early version of cmake had these on by default and we saw Joshua failures as a result). So I fear these need to be removed for now. The problem is that fdb never aligns memory but the compiler will emit some avx instructions that assume that memory is aligned.

Reply:
Uggh.  It really shouldn't do that.  According to this thread, the compiler should just emit unaligned variants. Newer processors don't have a performance penalty for aligned accesses that use the unaligned instructions:

https://software.intel.com/en-us/forums/intel-isa-extensions/topic/752392

Anyway, thanks for the heads up.  Do you remember which parts of the system were bitten by this?  I'll see if there's some way to force g++ to do the right thing.

Reply:
Hmm.. According to this:

https://gcc.gnu.org/projects/tree-ssa/vectorization.html#Alignment

the vectorizer added support for unaligned accesses in 4.0.  We're on a newer release than that.  How long ago did you hit the AVX/SSE issues?

Reply:
Last time we tried was some time after 6.2 was released - so this was either gcc 7 or gcc 8. 

@tclinken is currently working on memory alignment and he started with aligned Arenas. So maybe this will already be good enough. It could also be that this problem goes away after we move to clang.

But even if this does pass Joshua now, I think we should wait until the 6.3 branch was cut before we merge the PR in this state (unless you compile in a separate library and link statically)

Reply:
Btw: There's also a [comment](https://github.com/apple/foundationdb/blob/f329164fb4dbfe1e69251dfc45df58a22470f7ce/cmake/ConfigureCompiler.cmake#L178) in the code where we removed it that explains the problem. This would also be where these flags should be enabled again

Reply:
I looked at the bug referenced by the comment, and it seems the only issue was that ubsan was complaining.  Unaligned accesses are OK on X86 in general, and it looks like g++ (at least) only uses instructions that require alignment if it can prove the data in question is aligned.

I wonder if it was simply (correctly) pointing out that the code isn't guaranteed to work on all architectures.  If so, that's fine, and fixing it would make it easier to port FDB, but it has nothing to do with whether we should turn these flags on.

I found one report that claims it assumes that `__restrict` pointers are aligned, which is questionable behavior, but also understandable, and easy to work around. 

Did we actually hit bus errors due to unaligned access on x86?  I haven't been able to find any crashes by testing on my end.

Reply:
All I can tell you is that we ran Joshua with these flags enabled and we saw failures. At the time we didn't debug this. But instead of discussing this here it is probably easier to just send a build to Joshua and see what happens...